src/site/content/en/blog/extending-workbox/index.md

```diff
@@ -0,0 +1,379 @@
+[stale-while-revalidate](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#stale-while-revalidate),
+where a cached response is used to respond to a request immediately, while the
+cache is also updated so that it's fresh the next time around. They also include
+[network-first](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#network_first_network_falling_back_to_cache),
+falling back to the cache when the network is unavailable, and a few more.
+
+## Custom strategies
+
+But what if you wanted to go beyond those common caching scenarios? Let's cover
+writing your own custom caching strategies.
+[Workbox v6](https://github.com/GoogleChrome/workbox/releases/tag/v6.0.0) offers
+a new `Strategy` base class that sits in front of lower-level APIs, like
+[Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) and
+[Cache Storage](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage).
+You can extend the `Strategy` base class, and then implement your own logic in
+the `_handle()` method.
+
+### DedupeNetworkFirst
+
+For instance, imagine that you want to implement a strategy that can handle
+multiple, simultaneous requests for the same URL by deduplicating them. A copy
+of the response is then used to fulfill all of the in-flight requests, saving
+bandwidth that would otherwise be wasted.
+
+Here's the code you can use to implement that, by extending the `NetworkFirst`
+strategy (which itself extends the `Strategy` base):
+
+```javascript
+// See https://developers.google.com/web/tools/workbox/guides/using-bundlers
+import {NetworkFirst} from 'workbox-strategies';
+
+class DedupeNetworkFirst extends NetworkFirst {
+  constructor(options) {
+    super(options);
+    // This maps inflight requests to response promises.
+    this._requests = new Map();
+  }
+
+  // _handle is the standard entry point for our logic.
+  async _handle(request, handler) {
+    let responsePromise = this._requests.get(request.url);
+
+    if (responsePromise) {
+      // If there's already an inflight request, return a copy
+      // of the eventual response.
+      const response = await responsePromise;
+      return response.clone();
+    } else {
+      // If there isn't already an inflight request, then use
+      // the _handle() method of NetworkFirst to kick one off.
+      responsePromise = super._handle(request, handler);
+      this._requests.set(request.url, responsePromise);
+      try {
+        const response = await responsePromise;
+        return response.clone();
+      } finally {
+        // Make sure to clean up after a batch of inflight
+        // requests are fulfilled!
+        this._requests.delete(request.url);
+      }
+    }
+  }
+}
+```
+
+{% Aside %}
+This code assumes that all requests for the same URL can be
+satisfied with the same response, which won't always be the case if cookies or
+session state information comes into play.
+{% endAside %}
+
+### CacheNetworkRace
```

Comment:
```suggestion
### Create a race between the cache and network with CacheNetworkRace
```
Not sure how to summarize this section… this suggestion is probably good enough (unless you've got something better)